src/hashrate.rs

```diff
@@ -0,0 +1,74 @@
+use std::time::{Duration, Instant};
+use colored::*;
+
+
+pub fn get_unit(hashrate: f32) -> (f32, String) {
+    let mut hashrate = hashrate as f32;
+    let mut unit = "";
+    let units = vec!["k", "M", "G", "T", "P", "E", "Z", "Y"];
+    for u in units.iter() {
+        if hashrate > 1000.0 {
+            hashrate /= 1000.0;
+            unit = u;
+        } else {
+            break;
+        }
+    }
+    (hashrate, format!("{}H/s", unit))
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct Hashrate {
```

Comment:
I say, define two types. 
struct Hashrate(f32);

and WorkerHashrate{
 value:Hashrate,
 ...
}

and sum of  WorkerHashrates would be a Hashrate
This way, you won't need to assign id:0 for the total-hashrate

Reply:
I did not understand this (: